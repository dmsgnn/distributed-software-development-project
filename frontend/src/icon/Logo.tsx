import { Icon, IconProps } from '@chakra-ui/react';
import React, { FC } from 'react';

export const LogoSvg: FC<IconProps> = (props) => {
	return (
		<Icon color="primary.500" viewBox="0 0 38 39" {...props}>
			<path d="M0 .5h19a19 19 0 1 1-19 19V.5Z" fill="currentColor" />
		</Icon>
	);
};
