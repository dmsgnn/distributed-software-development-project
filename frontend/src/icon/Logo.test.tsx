import { axe } from 'jest-axe';

import { act, render } from 'test-utils';
import { LogoSvg } from './Logo';

describe('Logo', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<LogoSvg />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<LogoSvg />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});

		expect(results).toHaveNoViolations();
	});
});
