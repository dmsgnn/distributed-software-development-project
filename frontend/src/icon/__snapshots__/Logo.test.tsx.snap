// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should matches snapshot 1`] = `
<DocumentFragment>
  <svg
    class="chakra-icon css-q5208i"
    focusable="false"
    viewBox="0 0 38 39"
  >
    <path
      d="M0 .5h19a19 19 0 1 1-19 19V.5Z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;
